.github/workflows/test.yml

```diff
@@ -1,13 +1,67 @@
-name: Test
+name: test
 
 on:
+  pull_request_target:
   push:
     branches:
       - main
-  pull_request:
 
 jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: chartboost/ruff-action@v1
+  deploy:
+    needs: [ruff]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKERHUB_RPC_REPO }}
+          tags: |
+            type=ref,event=tag
+            type=ref,event=pr
+            type=raw,value=latest,enable={{is_default_branch}}
+          labels: |
+            maintainer=bitcoindevproject
+            org.opencontainers.image.title=warnet-rpc
+            org.opencontainers.image.description=Warnet RPC server
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          file: src/templates/rpc/Dockerfile_rpc
+          platforms: linux/amd64,linux/arm64
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=registry,ref=${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKERHUB_RPC_REPO }}:latest
+          cache-to: type=inline
+      - name: Create tag name file
+        run: echo "${{ steps.meta.outputs.tags }}" > tagname.txt
```

Comment:
this has to be a file? it cant just be exported like the other variables `{{ stepa.meta.output.tags }}` etc?

Reply:
This is an artifact of when the jobs were in different workflows. It does not need a file now